utopian-io/v2.utopian.io
# [FEATURE] Google sign in

### My PR doesn't concern a Jira task
Implements Google sign in:

- Sign in with a Google account
- Create account from Google sign in
- Cross sign in between auth providers
---
## New env vars:

### packages/auth
* GOOGLE_CLIENT_ID=''
* GOOGLE_REDIRECT_URL=http://localhost:8081

### packages/api
* GOOGLE_CLIENT_ID=''
* GOOGLE_CLIENT_SECRET=''
* GOOGLE_REDIRECT_URL=http://localhost:8081

### **Test plan**

_Sign up with Google_ 

* Delete your user account or test with a blank database
* Go to localhost:8080 and click on Sign In
* In the auth app, click on Sign in with Google
* Accept the request and you'll be redirected to the signup page
* Create a username 
* You should be redirected to the main app

_Sign in with Google_ 
* After creating your account, log out
* Click on Sign In again
* Click on Sign In with Google
* You should be redirected to the main app

_Sign in with Github account with same email_
* Make sure your github account has the same email that your Google account
* Sign out and sign in with Github
* You should be redirected to the main app (not to the sign up page)
